src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [ ];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, action.extraData);
+        result.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties' :
+        for (const item of action.keysToRemove) {
+          delete stateCopy[item];
+        }
+        result.push({ ...stateCopy });
+        break;
+
+      case 'clear' :
+        for (const key of Object.keys(stateCopy)) {
+          delete stateCopy[key];
+        }
+        result.push({ ...stateCopy });
+        break;
+    }
```

Comment:
Follow the checklist
![image](https://github.com/mate-academy/js_stateful-clones/assets/99403198/fbcdde17-cd60-4e72-9f98-7b78f678fe80)